packages/devtools_app/lib/src/shared/console/widgets/evaluate.dart

```diff
@@ -134,16 +134,17 @@ class ExpressionEvalFieldState extends State<ExpressionEvalField>
     }
   }
 
-  void _handleSearch() async {
+  void _handleSearchTermSelected() {
+    _autoCompleteController.clearCurrentSuggestion();
+  }
+
+  Future<void> _handleSearch() async {
     final searchingValue = _autoCompleteController.search;
 
     _autoCompleteController.clearCurrentSuggestion();
 
     if (searchingValue.isNotEmpty) {
-      if (_autoCompleteController.selectTheSearch) {
-        _autoCompleteController.resetSearch();
```

Comment:
Resetting the search term here was the crux of the bug. This would clear whatever the user had already inputted.